File: sqlparser/src/sql/SqlConstantColumn.java
//@
package cdata.sql;
//@
/*#
using RSSBus.core;
using RSSBus;
namespace CData.Sql {
#*/

public final class SqlConstantColumn extends SqlColumn implements ISqlElement {
  private final SqlValueExpression expr;

  public SqlConstantColumn(SqlValueExpression val) {
    super(null, null);
    expr = val;
  }

  public SqlConstantColumn(String alias, SqlValueExpression val) {
    super(alias, alias);
    expr = val;
  }

  public /*#override#*/ SqlExpression getExpr() {
    return this.expr;
  }

  public /*#override#*/ Object clone() {
    SqlConstantColumn obj = new SqlConstantColumn(this.getAlias(), this.expr);
    obj.copy(this);
    return obj;
  }

  public /*#override#*/ boolean equals(String name) {
    return false;
  }

  public /*#override#*/ boolean isEvaluatable() {
    return expr != null;
  }

  public /*#override#*/ SqlValue evaluate() {
    if (expr != null) {
      return expr.evaluate();
    }
    return SqlValue.getNullValueInstance();
  }

  public /*#override#*/ boolean hasAlias() {
    if (getAlias() != null) {
      return true;
    } else {
      return false;
    }
  }
}

File: sqlparser/src/sql/csgen/SqlConstantColumn.cs
using System;
using System.Collections;
using System.IO;
using System.Threading;
using System.Net;
using System.Net.Sockets;
using System.Text;
using RSSBus.core.j2cs;



using RSSBus.core;
using RSSBus;
namespace CData.Sql {


public sealed class SqlConstantColumn : SqlColumn , ISqlElement {
  private readonly SqlValueExpression expr;

  public SqlConstantColumn(SqlValueExpression val) : base(null, null) {
    expr = val;
  }

  public SqlConstantColumn(string alias, SqlValueExpression val) : base(alias, alias) {
    expr = val;
  }

  public override SqlExpression GetExpr() {
    return this.expr;
  }

  public override Object Clone() {
    SqlConstantColumn obj = new SqlConstantColumn(this.GetAlias(), this.expr);
    obj.Copy(this);
    return obj;
  }

  public override bool Equals(string name) {
    return false;
  }

  public override bool IsEvaluatable() {
    return expr != null;
  }

  public override SqlValue Evaluate() {
    if (expr != null) {
      return expr.Evaluate();
    }
    return SqlValue.GetNullValueInstance();
  }

  public override bool HasAlias() {
    if (GetAlias() != null) {
      return true;
    } else {
      return false;
    }
  }
}
}

